### PR TITLE
Add npm run lint

### PR DESCRIPTION
**Pull request by koba04**
This PR is for adding `npm run lint`, which is to lint code in `src` and `test` directories.
In addition to that, I've fixed a lint error in `test`.

### CODE REVIEW

File: test/CSSTransitionGroup-test.js [hunk 1]
```diff
@@ -298,7 +298,9 @@ describe('CSSTransitionGroup', () => {
       state = { showChild: true };
 
       componentDidMount() {
-        this.setState({ showChild: false });
+        setTimeout(() => {
+          this.setState({ showChild: false });
+        });
```
Comment (jquense) [inline test/CSSTransitionGroup-test.js]: ?

Comment (koba04) [inline test/CSSTransitionGroup-test.js]: This diff is for to fix `react/no-did-mount-set-state`...
I guess there are better ways...

Comment (jquense) [inline test/CSSTransitionGroup-test.js]: lets just disable that rule (i have it disabled in a branch actually)

Comment (koba04) [inline test/CSSTransitionGroup-test.js]: Disabled `react/no-did-mount-set-state`

**Comment by jquense** [comment]
thanks!

**Comment by koba04** [comment]
Thanks!

**Pull request merged**